src/Sulu/Component/Content/Document/Subscriber/ResourceSegmentSubscriber.php

```diff
@@ -57,7 +57,9 @@ public function doHydrate(AbstractMappingEvent $event)
     {
         $document = $event->getDocument();
         $property = $this->getResourceSegmentProperty($document);
-        $segment = $document->getStructure()->getProperty($property->getName())->getValue();
+
+        $locale = $this->inspector->getOriginalLocale($document);
```

Comment:
@dantleech do you have a better idea here? we should load the the url in the original locale! /cc @danrot